blocks/graph.py

```diff
@@ -161,8 +162,37 @@ def replace(self, replacements):
             substitutes.
 
         """
-        return ComputationGraph(theano.clone(self.outputs,
-                                             replace=replacements))
+        # Due to theano specifics we have to make one replacement in time
+        replacements = OrderedDict(replacements)
+        # Replacements with previous replacements
```

Comment:
Please, explain more what `replacement_cur` and `replacement_keys` actually contain. Be more generous when you write comments to non-trivial code.